src/scripts/list-users.test.ts

```diff
@@ -0,0 +1,176 @@
+    expect(actual).toContain('test@localhost');
+    expect(actual).toContain('test user');
+    expect(actual).toContain('ACTIVE');
+
+    // And we also expect getGroup to have been called
+    expect(groupService.getGroup).toBeCalledWith(test.group.id);
+  });
+
+  it('fails when attempting to list users within an existing group with error', async () => {
+    // Given a user service that fails to retrieve a list of users
+    const userService: UserService = {
+      ...test.baseUserService(),
+      listUsersInGroup: () => TE.left('No users.'),
+    };
+
+    // And a group service that succsessfully retreives a group
+    const groupService: GroupService = {
+      ...test.baseGroupService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      getGroup: jest.fn((_group) => TE.right(O.some(test.group))),
+    };
+
+    // When we attempt to list the users from the client
+    const result = await usersInGroup(
+      userService,
+      groupService,
+      test.group.id
+    )();
+
+    // Then we should have a left
+    expect(result).toEqualLeft('No users.');
+
+    // But getGroup was called
+    expect(groupService.getGroup).toBeCalledWith(test.group.id);
+  });
+
+  it('fails when attempting to list users from a non-existent group', async () => {
+    // Given a user service that succsessfully retrieves a list of users (which should be impossible if the group doesn't exist)
+    const userService: UserService = {
+      ...test.baseUserService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      listUsersInGroup: jest.fn((_group) => TE.right([test.user])),
+    };
+
+    // And a group service that fails to find the group
+    const groupService: GroupService = {
+      ...test.baseGroupService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      getGroup: (_group) => TE.left('No group.'),
+    };
+
+    // When we attempt to list the users from the client
+    const result = await usersInGroup(
+      userService,
+      groupService,
+      test.group.id
+    )();
+
+    // Then we should have a left
+    expect(result).toEqualLeft('No group.');
+
+    // And listUsersInGroup was never called
+    expect(userService.listUsersInGroup).not.toHaveBeenCalled();
+  });
+
+  it('fails when attempting to list users from a none group', async () => {
+    // Given a user service that succsessfully retrieves a list of users (which should be impossible if the group doesn't exist)
+    const userService: UserService = {
+      ...test.baseUserService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      listUsersInGroup: jest.fn((_group) => TE.right([test.user])),
+    };
+
+    // And a group service that fails to find the group, this time returning none instead of an error
```

Comment:
Technically, the group service does not fail.

Reply:
The wording has been changed